### PR TITLE
[859] Update content on visa question

### PR DESCRIPTION
**Pull request by avinhurry**
### Context

Updating the visa sponsorship question on Find in an effort to reduce rejection rates.

### Before

image

### After

image

### Guidance to review

Link to the page in the review app: https://find-review-3906.test.teacherservices.cloud/visa-status?age_group=secondary&l=2&subjects%5B%5D=&subjects%5B%5D=C1

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [x] Inform data insights team due to database changes

### CODE REVIEW

**Comment by avinhurry** [comment]
[Implement improved wording on our visa sponsorship question in Find](https://trello.com/c/P8ZjLv2V/859-implement-improved-wording-on-our-visa-sponsorship-question-in-find)

**Review by MaeveRoseveare — COMMENTED** [review]
Content looks good, but I think we need to change the logic of when the 'visa sponsorship' filter is selected because the Yes/No question has changed.

Now:
if I select 'Yes' I need visa sponsorship - the 'visa sponsorship' filter should be selected
if I select 'No' I do not need visa sponsorship - the 'visa sponsorship' filter should **NOT** be selected

At the moment it's the opposite.

**Review by MaeveRoseveare — APPROVED** [review]
Can't delete this comment... Ignore this.

File: app/views/find/result_filters/_visa_filter.html.erb
Comment (tomas-stefano) [inline app/views/find/result_filters/_visa_filter.html.erb:9]: The can sponsor visa checked is expexted a negation of `visa_checked?`. Need to understand a little, can you clarify, please?

Comment (avinhurry) [reply app/views/find/result_filters/_visa_filter.html.erb:9]: It's this: https://github.com/DFE-Digital/publish-teacher-training/pull/3906#pullrequestreview-1704003244

image

Does that make sense?

File: app/views/find/search/visa_status/new.html.erb
Comment (MylesJarvis) [inline app/views/find/search/visa_status/new.html.erb:32]: @MaeveRoseveare should the h2 class here be "govuk-heading-m"?

I think the fact that the H1 also uses "govuk-heading-l" means they end up being the same size?

image

File: app/views/find/result_filters/_visa_filter.html.erb
Comment (tomas-stefano) [reply app/views/find/result_filters/_visa_filter.html.erb:9]: Sorry I should had phrase better. the code `checked: !@filters_view.visa_checked?` is confusing for me, because the #visa_checked? method returns true we can expect the filter to be checked, but the code is showing the inverse, if `#visa_checked?` returns true the checked option will be false.

Comment (MylesJarvis) [reply app/views/find/result_filters/_visa_filter.html.erb:9]: I've just given @avinhurry a couple of comments on this filter - it appears it's showing results the wrong way round at present. That is to say:
- If I say `Yes` to the Q: Do you need visa sponsorship, I see all courses, including those that don't sponsor visas
- If I say `no` to the Q: Do you need visa sponsorship, I see limited courses, only ones that sponsor visas.

**Comment by MylesJarvis** [comment]
Just adding this in the full thread so we don't lose sight of it - needs fixing before we merge:
https://github.com/DFE-Digital/publish-teacher-training/pull/3906#discussion_r1377936488

File: app/views/find/search/visa_status/new.html.erb
Comment (MylesJarvis) [inline app/views/find/search/visa_status/new.html.erb:35]:
```suggestion
        <%= f.govuk_radio_button :visa_status, "false", label: { text: "No" } %>
```

File: app/views/find/search/visa_status/new.html.erb
Comment (MylesJarvis) [inline app/views/find/search/visa_status/new.html.erb:34]:
```suggestion
        <%= f.govuk_radio_button :visa_status, "true", label: { text: "Yes" }, link_errors: true %>
```

**Review by MylesJarvis — CHANGES_REQUESTED** [review]
Adding these in the hope it blocks merging until fixed.

**Review by MaeveRoseveare — APPROVED** [review]
This is working now and the questions + filter checked/unchecked is the right way around!

File: spec/features/find/search/editing_a_search_spec.rb
Comment (tomas-stefano) [inline spec/features/find/search/editing_a_search_spec.rb:36]: should we add a step `then_the_visa_filter_should_be_checked` or something related?

File: spec/features/find/search/across_england/secondary_spec.rb
Comment (tomas-stefano) [inline spec/features/find/search/across_england/secondary_spec.rb:160]: should we add a step `then_the_visa_filter_should_not_be_checked` or something related?

File: spec/features/find/search/editing_a_search_spec.rb
Comment (avinhurry) [reply spec/features/find/search/editing_a_search_spec.rb:36]: Great idea. 

Done here: https://github.com/DFE-Digital/publish-teacher-training/pull/3906/commits/779374366a8669ad53bf8b95f691bc3c7af780ae

File: spec/features/find/search/across_england/secondary_spec.rb
Comment (avinhurry) [reply spec/features/find/search/across_england/secondary_spec.rb:166]: Yep, also done here: https://github.com/DFE-Digital/publish-teacher-training/pull/3906/commits/c6753463b1483e05d89c47a335336de538a52a82

**Review by tomas-stefano — APPROVED** [review]

**Pull request merged**
